datafusion/physical-plan/src/lib.rs

```diff
@@ -852,6 +852,30 @@ pub fn spill_record_batches(
     Ok(writer.num_rows)
 }
 
+/// Spill the `RecordBatch` to disk as smaller batches
+/// split by `batch_size`
+/// Return `total_rows` what is spilled
+pub fn spill_record_batch_by_size(
+    batch: RecordBatch,
+    path: PathBuf,
+    schema: SchemaRef,
+    batch_size: usize,
+) -> Result<usize, DataFusionError> {
```

Comment:
Exactly the idea behind is to make sub batches to help the consumer reading data using less memory. The same approach we use in `row_hash.rs`
Ideally to be honest is to return stream `SendableBatchRecordStream` but this will require more efforts as the SMJ is not async for now.